tests/stdlib/tonce.nim

```diff
@@ -0,0 +1,34 @@
+discard """
+  cmd: "nim c -r --threads:on $options $file"
+  matrix: "-d:caseA; -d:caseB"
+"""
+import std/once
+
+when defined(caseA):
+  block:
+    var thr: array[0..4, Thread[void]]
+    var block1: Once
+    var count = 0
+    initOnce(block1)
+    proc threadFunc() {.thread.} =
+      for i in 1 .. 10:
+        once(block1):
+          inc count
+    for i in 0..high(thr):
```

Comment:
more idiomatic:
```nim
for t in mitems(thr):
  createThread(t, threadFunc)
```
(and Example code in https://github.com/nim-lang/Nim/blob/73299b048f57838d83a3a8159909386a0905e553/lib/system/threads.nim#L41 should be updated as well in future work)

ditto below